src/realm/db.cpp

```diff
@@ -1350,9 +1365,13 @@ bool DB::compact(bool bump_version_number, util::Optional<const char*> output_en
         top_ref = m_alloc.attach_file(m_db_path, cfg);
         m_alloc.init_mapping_management(info->latest_version_number);
         info->number_of_versions = 1;
-        SharedInfo* r_info = m_reader_map.get_addr();
-        size_t file_size = m_alloc.get_baseline();
-        r_info->init_versioning(top_ref, file_size, info->latest_version_number);
+        size_t logical_file_size = 24;
+        if (top_ref) {
+            Array top(m_alloc);
+            top.init_from_ref(top_ref);
+            logical_file_size = Group::get_logical_file_size(top);
+        }
+        m_version_manager->init_versioning(top_ref, logical_file_size, info->latest_version_number);
```

Comment:
Is this actually valid? Should we go through and validate that all counts are 0 before starting the compact?

Reply:
It should be ensured by the checks earlier in the same function. ReadLocks are taken for transaction objects, and we check that only a single transaction object exists.